sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java

```diff
@@ -2405,6 +2407,11 @@ void visitSort(LogicalSort sort) {
         RelNode input = sort.getInput();
         DBSPOperator opInput = this.getOperator(input);
         if (this.options.languageOptions.ignoreOrderBy && sort.fetch == null) {
+            this.compiler.reportWarning(node.getPositionRange(), "ORDER BY is ignored",
+                    "ORDER BY clause " +
+                    (this.currentView != null ?
+                            "producing view " + Utilities.singleQuote(this.currentView.relationName) + " " : "") +
+                    "will be ignored");
```

Comment:
```suggestion
                    "will be ignored, you can pair the `ORDER BY` with a `LIMIT n` clause for it to be incrementally evaluated");
```

Reply:
I don't think we guarantee an order even if you use LIMIT.

Reply:
isn't this top-k then?

Reply:
it is topk, but it's not ordered topk

Reply:
this may be good enough for some use-cases, it's still better than giving no suggestion on what to do

Reply:
what to do to achieve what?
the result in both cases has the right rows, just not ordered
and we can build a special DBSP node to keep a materialized view in order, thus making this warning obsolete #2833

Reply:
in addition if you can link to somewhere in the docs where it is explained why this doesn't work might be useful too

Reply:
the funny thing is that the materialized views we maintain *are* actually sorted, so we do all the required work. But they are sorted using Ord, and not the user-specified order.

Reply:
the doc is issue #2834, a PR for that can include a link in this warning

Reply:
>thus making this warning obsolete

then you can adjust the warning again, at least lets leave the hint with the adhoc queries if you don't like the limit suggestion. but I'd argue order-by + limit is still a useful suggestions

Reply:
a suggestion to achieve what? the result will still not be ordered.
as I said, both versions are wrong in the exact same way

Reply:
maybe I should emit the warning in that case as well

Reply:
> as I said, both versions are wrong in the exact same way

when starting out from the original query of just order-by making it top-k of a smaller range is arguably better because it will actually give you the top-k elements from the original table if you're interested in that. So while not sorted this suggestion can be useful certain use-cases.